apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/components/sidebar-modal-opener.js

```diff
@@ -0,0 +1,108 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { Component } from '@wordpress/element';
+import { withSelect } from '@wordpress/data';
+import { Button, Modal } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { __ } from '@wordpress/i18n';
+/**
+ * Internal dependencies
+ */
+import { PageTemplatesPlugin } from '../index';
+import TemplateSelectorItem from './template-selector-item';
+import replacePlaceholders from '../utils/replace-placeholders';
+/* eslint-enable import/no-extraneous-dependencies */
+
+class SidebarModalOpener extends Component {
+	state = {
+		isTemplateModalOpen: false,
+		isWarningOpen: false,
+	};
+
+	toggleTemplateModal = () => {
+		this.setState( { isTemplateModalOpen: ! this.state.isTemplateModalOpen } );
+	};
+
+	toggleWarningModal = () => {
+		this.setState( { isWarningOpen: ! this.state.isWarningOpen } );
+	};
+
+	getLastTemplateUsed = ( { templateUsedSlug, templates } = this.props ) => {
+		if ( ! templateUsedSlug || templateUsedSlug === 'blank' ) {
+			// If no template used or 'blank', preview any other template (1 is currently 'Home' template).
+			return templates[ 1 ];
+		}
+		return templates.find( temp => temp.slug === templateUsedSlug );
+	};
+
+	render() {
+		const { slug, title, preview, previewAlt } = this.getLastTemplateUsed();
+		const { templates, vertical, segment, siteInformation } = this.props;
+
+		return (
+			<div className="sidebar-modal-opener">
+				<TemplateSelectorItem
+					id="sidebar-modal-opener__last-template-used-preview"
+					value={ slug }
+					label={ replacePlaceholders( title, siteInformation ) }
+					staticPreviewImg={ preview }
+					staticPreviewImgAlt={ previewAlt }
+					onSelect={ this.toggleWarningModal }
+				/>
+
+				<Button
+					isPrimary
+					onClick={ this.toggleWarningModal }
+					className="sidebar-modal-opener__button"
+				>
+					{ __( 'Change Layout' ) }
+				</Button>
+
+				{ this.state.isTemplateModalOpen ? (
```

Comment:
I common pattern we use in Calypso would look something like: `this.state.isTemplateModalOpen && <Component />`. That way you don't have to return `null` explicitly.

Reply:
I like it!  I forgot that was a thing.  Updated.